### PR TITLE
Image tags

### LINKED ISSUE #7: Image Tags
#### As a user I want to add tags to images.

Acceptance criteria:
- [x] When I add a new image, there is a form field to add tags.
- [x] After I've saved an image with tags, I see those tags displayed with the
  associated image on the index page.
- [x] My application uses the acts-as-taggable-on gem to provide the tagging
  functionality.

Reference material:
- https://github.com/mbleigh/acts-as-taggable-on

Dependencies:
- Image Index

### PR DESCRIPTION
**Pull request by nicklyz**
Closes #7 

Added gem acts_as_taggable. 
- Added one field to the form
- Created tag tables

### CODE REVIEW

File: db/migrate/20160627173054_acts_as_taggable_on_migration.acts_as_taggable_on_engine.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# This migration comes from acts_as_taggable_on_engine (originally 1)
+class ActsAsTaggableOnMigration < ActiveRecord::Migration
+  def self.up
```
Comment (weixing2014) [inline db/migrate/20160627173054_acts_as_taggable_on_migration.acts_as_taggable_on_engine.rb]: I think `def up` is enough.

File: test/controllers/images_controller_test.rb [hunk 2]
```diff
@@ -7,20 +7,27 @@ class ImagesControllerTest < ActionController::TestCase
 
   test 'should_save_image' do
     assert_difference('Image.count', 1) do
-      post :create, link: @image.link
+      post :create, image: { link: @image.link }
     end
     assert_redirected_to root_url
   end
 
   test 'should_reject_invalid_url' do
```
Comment (weixing2014) [inline test/controllers/images_controller_test.rb]: In the controller tests, you might want to test one case only in each test. You can split this test into several different tests.

File: db/migrate/20160627173054_acts_as_taggable_on_migration.acts_as_taggable_on_engine.rb (hunk 1 above)
Comment (nicklyz) [inline db/migrate/20160627173054_acts_as_taggable_on_migration.acts_as_taggable_on_engine.rb]: > I think def up is enough.

I think this is generated by the gem. Btw what is the difference between 'self.up' and 'up'?

File: db/migrate/20160627173056_add_taggings_counter_cache_to_tags.acts_as_taggable_on_engine.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# This migration comes from acts_as_taggable_on_engine (originally 3)
+class AddTaggingsCounterCacheToTags < ActiveRecord::Migration
+  def self.up
+    add_column :tags, :taggings_count, :integer, default: 0
+
+    ActsAsTaggableOn::Tag.reset_column_information
```
Comment (weixing2014) [inline db/migrate/20160627173056_add_taggings_counter_cache_to_tags.acts_as_taggable_on_engine.rb]: You might not want to use `Model` in rails migration. Check this out https://coderwall.com/p/zav1dg/using-models-in-rails-migrations

**Pull request merged**